### DISCUSSION TITLE
How to declare a variable of a Prisma model?

### DISCUSSION THREAD

**Discussion by namleasia** [prisma/prisma#8246] [Q&A]
TypeScript.

I have this model:
```javascript
model User {
  id        Int      @id
  email     String   @unique
  name      String?
  posts     Post[]
}
```
How to declare a variable of this model?
I tried this, but TypeScript not allow:
```javascript
const user: User = new User();
```
TypeScript says: **'User' only refers to a type, but is being used as a value here.ts(2693)**

Please help!

**Comment by ryands17** [comment]
@namleasia 👋 

You can directly create an object from the type. Prisma doesn't have classes so you can write it like this:

```ts
const user: User = {
  email: 'email',
  name: 'name'
}
```

**Reply by namleasia** [reply]
Thank you Ryands, it's ok, but I don't want to repeat my self to rewrite these properties everytime. (my model has many properties).

**Reply by ryands17** [reply]
I didn't get your point. Can you elaborate?

**Reply by namleasia** [reply]
I don't want to write properties: id, email, name... when declaring variable like your suggestion.
Because these properties are already in Prisma model.
Is there a way to do something like:
```javascript
const user: User = new User(); // User is Prisma model
```
Then I have all properties (id, email, name...) in **user** variable.

**Reply by ryands17** [reply]
Could you explain your use case for this? This is not possible as Prisma is not a class based ORM so I can suggest an alternative depending on what you want to do.

**Reply by namleasia** [reply]
No special use case, just because my Prisma model has many properties (44 properties), so I don't want to write properties when declare variable.
I am using Nestjs (TypeScript) and Prisma.

Thank you so much!

Another Issue: https://github.com/prisma/prisma/issues/8238

**Comment by yoni2356** [comment]
I have a similar use-case, 
I am trying to create an instance of an entity - although I do not want to "DB Insert" the instance per-se. 
I want to create it in server memory only - and return it.

Right now my main problem is (id default autoincrement),
Since to create an instance in memory, its type requires an id,
I can provide 0, yet I was looking for a better solution.

Currently trying Partial(MyEntity) - (I know these arent < and >, the forum doesn't show them), 
but lacking knowledge on how to handle nested relations (one to many array-like), 
that each of the related items require "Partial" aswell.

**Reply by ryands17** [reply]
@yoni2356 👋 

Prisma is not a class based ORM so you cannot create instances, we do have a feature request for the same here so it would be great if you could add a 👍 to this so that we know the priority for this: https://github.com/prisma/prisma/issues/5998

**Reply by yoni2356** [reply]
Thank you for the reply, appreciated!
Shall add a like asap :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
